shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,171 @@
+            if (!processorStatus.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            return UpdateControl.updateStatus(managedProcessor).rescheduleAfter(executorPollIntervalMilliseconds);
+        } else if (!processorStatus.isConditionTypeTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE)) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            processorStatus.markConditionTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE);
+        }
+
+        LOGGER.info("Managed Processor: '{}' in namespace '{}' is ready",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        // Only issue a Status Update once.
+        // This is a work-around for non-deterministic Unit Tests.
+        // See https://issues.redhat.com/browse/MGDOBR-1002
+        if (!managedProcessor.getStatus().isReady()) {
+            metricsService.onOperationComplete(managedProcessor, MetricsOperation.CONTROLLER_RESOURCE_PROVISION);
+            processorStatus.markConditionTrue(ConditionTypeConstants.READY);
+            // Notify Manager is Ready
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        return UpdateControl.noUpdate();
+    }
+
+    private boolean isTimedOut(ManagedProcessorStatus status) {
+        Optional<Date> lastTransitionDate = status.getConditions()
+                .stream()
+                .filter(c -> Objects.nonNull(c.getLastTransitionTime()))
+                .reduce((c1, c2) -> c1.getLastTransitionTime().after(c2.getLastTransitionTime()) ? c1 : c2)
+                .map(Condition::getLastTransitionTime);
+
+        if (lastTransitionDate.isEmpty()) {
+            return false;
+        }
+
+        ZonedDateTime zonedLastTransition = ZonedDateTime.ofInstant(lastTransitionDate.get().toInstant(), ZoneOffset.UTC);
+        ZonedDateTime zonedNow = ZonedDateTime.now(ZoneOffset.UTC);
+        return zonedNow.minus(Duration.of(executorTimeoutSeconds, ChronoUnit.SECONDS)).isAfter(zonedLastTransition);
+    }
+
+    @Override
+    public DeleteControl cleanup(ManagedProcessor ManagedProcessor, Context context) {
+        LOGGER.info("Deleted ManagedProcessor: '{}' in namespace '{}'", ManagedProcessor.getMetadata().getName(), ManagedProcessor.getMetadata().getNamespace());
+
+        // Linked resources are automatically deleted
+
+        metricsService.onOperationComplete(ManagedProcessor, MetricsOperation.CONTROLLER_RESOURCE_DELETE);
+        // notify manager it's been deleted
+
+        return DeleteControl.defaultDelete();
+    }
+
+    @Override
+    public Optional<ManagedProcessor> updateErrorStatus(ManagedProcessor ManagedProcessor, RetryInfo retryInfo, RuntimeException e) {
```

Comment:
Similar comment here as to the timeout. We need to discuss how we're going to report errors via Conditions. This is again using the `Ready` Condition which we don't have on the `ManagedBridge`. In the `ManagedBridgeController` we're not doing anything with this yet, so up to you if want to add it here.